airflow/operators/python.py

```diff
@@ -178,6 +178,8 @@ class BranchPythonOperator(PythonOperator, SkipMixin):
 
     def execute(self, context: Dict):
         branch = super().execute(context)
+        if branch is None:
+            raise AirflowException("Branch callable must return at least one task_id to follow")
```

Comment:
Kind of wondering whether we should check for all non-str values. Or even check the value is a valid task ID? Something like…

```python
if isinstance(branch, str):
    branches = [branch]
try:
    branches = list(branch)  # Check iterability.
except TypeError:
    raise AirflowException("Branch callable must return either a task ID or a list of IDs")
valid_task_ids = set(context["context"].task_ids)
if not any(b in valid_task_ids for b in branches):
    raise AirflowException("Branch callable must return at least one task ID to follow")
```

Reply:
yeah this is something I had in mind - added it in last commit
In general this is something we should have in skipmixin as it's logic shared by all branch operators.